Explicitly depend on WildFly-provided libraries

jasonaowen opened the pull request:
WildFly ships with several libraries, which are available to applications deployed to it at runtime. We also need those libraries to compile our application. Previously, we required a copy of WildFly to live in a particular directory to pick up those libraries. We shouldn't do that, though!

Instead, explicitly depend on the libraries we require, and use the [gradle `compileOnly` scope](https://blog.gradle.org/introducing-compile-only-dependencies) to indicate that these dependencies are available at runtime and should not be bundled into our application.

Along the way, fix a `@NotNull` vs `@Column(nullable = false)` mistake I made during our Hibernate migration, so as to avoid introducing an additional dependency.

I believe this is the last piece of issue #16!

---

I tested this by doing a clean build (which should pick up any missing dependencies), running the unit tests (`./gradlew {cms-business-model,cms-business-process,cms-portal-services,cms-web,services}:test`), deploying, and running the integration tests (`./gradlew integration-tests:{clean,test,aggregate}`).

---

Resolves #16 Manage sets of dependencies via Gradle or another tool

cecilia-donnelly commented on psm-app/build.gradle:
```diff
@@ -55,6 +55,10 @@ ext.libs = [
 ]
 
+ext.wildfly_libs = [
+    hibernate_core: 'org.hibernate:hibernate-core:5.0.10.Final',
```
Does this mean that we'll be pulling hibernate down from the Maven Repo or similar instead of from Wildfly?

cecilia-donnelly replied:
And, I suppose, does this make it easier to upgrade Wildfly?

jasonaowen commented:
>  How did you know which javax dependencies we have?

I deleted the references to the WildFly directories, looked at the resulting compile errors, added the references back in, used IntelliJ to figure out what libraries had the needed classes, and matched the version WildFly provided to the version on Maven Central.

Now that I write this out, though, it occurs to me this approach still ties us more tightly to WildFly than we should be - better to use the more-generic, non-JBoss libraries for, eg, the EJB API. That should allow us to switch to another application server more easily. I'll look into that a bit today.

jasonaowen commented:
Jenkins, retest this please.

jasonaowen commented:
Jenkins, retest this please.

jasonaowen commented:
Jenkins, retest this please.

jasonaowen commented:
Jenkins, retest this please.

jasonaowen commented:
Jenkins, retest this please.

jasonaowen commented:
Jenkins, retest this please.

jasonaowen commented:
Jenkins, retest this please.

jasonaowen commented:
For anyone following along, I apologize for the spam as I work on getting Jenkins configured correctly. This is a useful PR to test with, as it fixes the problem on master of needing WildFly installed. I suggest unsubscribing from notifications on this PR.

jasonaowen commented:
Jenkins, retest this please.

jasonaowen commented:
Jenkins, retest this please.
